A72-open-telemetry-tracing.md

```diff
@@ -0,0 +1,680 @@
+* Approver: [Eric Anderson](https://github.com/ejona86)
+* Reviewers: [Mark Roth](https://github.com/markdroth), [Doug Fawley](https://github.com/dfawley),
+[Feng Li](https://github.com/fengli79)
+* Status: Ready for Implementation
+* Implemented in: <language, ...>
+* Last updated: 2024-01
+* Discussion at: https://groups.google.com/g/grpc-io/c/e_ByaRmtJak
+
+# Abstract
+This proposal adds support for OpenTelemetry tracing and suggests migration
+paths away from OpenCensus tracing. Discussions include:
+* The API surface to enable and configure OpenTelemetry tracing.
+* Context propagation between a gRPC client and server.
+* Migration path from gRPC OpenCensus to OpenTelemetry, considering:
+  1) The cross-process concerns during migration.
+  2) In-binary migration for a gRPC involved software that has both OpenTelemetry and 
+    OpenCensus dependency.
+
+Note that stats and logging are out of scope.
+
+# Background
+This work aligns with the community consensus to switch to OpenTelemetry as the 
+next generation OpenCensus. The latter is no longer maintained after July 31, 2023.
+
+Currently, gRPC supports OpenCensus based tracing in its grpc-census plugin, or 
+alike. gRPC trace is built with its core stream tracer and interceptor 
+infrastructures. A gRPC client when intercepting the call creates a child span
+with the parent span from the current context, and further creates attempt spans
+upon stream creation for each attempt. [gRFC A45][A45] describes cross language 
+design considerations about creating a child span for each individual call attempt. 
+A gRPC server uses span ID from the incoming request header as a parent span to 
+maintain the parent child span relationship with the gRPC client. To propagate 
+span context over the wire, gRPC uses metadata (header name: `grpc-trace-bin`) 
+and OpenCensus's binary format for (de)serialization. The header name is unique 
+from other census library propagators to differentiate with the application’s 
+tracing instrumentation.
+
+### Related Proposals and Documents:
+* [gRFC L29: C++ API Changes for OpenCensus Integration][L29]
+* [gRFC A45: Exposing OpenCensus Metrics and Tracing for gRPC retry][A45]
+* [gRFC A66: OpenTelemetry Metrics][A66]
+
+# Proposal
+## gRPC OpenTelemetry Tracing API
+We will add tracing functions in grpc-open-telemetry plugin, along with OpenTelemetry
+metrics [gRFC A66][A66]. Internally, the tracing functionality will be implemented
+using existing gRPC infrastructure such as interceptors and stream tracers.
+The APIs to enable and configure OpenTelemetry tracing are different among 
+languages due to different underlying infrastructures.
+
+### Java
+In Java, it will be part of global interceptors, so that the interceptors are
+managed in a more sustainable way and user-friendly. As a prerequisite, the stream
+tracer factory API will be stabilized. OpenTelemetryModule will be created with
```

Comment:
We aren't going to stabilize the tracer API. And I don't see a reason we need to, here.